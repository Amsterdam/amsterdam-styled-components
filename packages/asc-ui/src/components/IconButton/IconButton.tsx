import React from 'react'
import { AscCore } from '@datapunt/asc-core'
import { Icon } from '../..'

type Props = {
  onClick?: React.MouseEventHandler<HTMLButtonElement>
}

const IconButton: React.FC<Props & AscCore.Button.Props> = ({
  children,
  onClick,
  ...otherProps
}) => (
  <AscCore.Button.IconButton onClick={onClick} {...otherProps} square>
    <Icon>{children}</Icon>
  </AscCore.Button.IconButton>
)

export default IconButton
