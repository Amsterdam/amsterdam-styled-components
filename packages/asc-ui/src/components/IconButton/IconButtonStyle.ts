import styled, { css } from '@datapunt/asc-core'
import { size } from 'polished'
import ButtonBaseStyle, {
  Props as ButtonBaseStyleProps,
} from '../Button/ButtonBaseStyle'
import IconStyle from '../Icon/IconStyle'

export type Props = {
  iconSize?: number
} & ButtonBaseStyleProps

export const IconButtonStyle = styled(ButtonBaseStyle)<Props>`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 5px;
  ${({ size: sizeProp }) =>
    sizeProp
      ? size(sizeProp)
      : css`
          width: 30px;
          height: 30px;
        `}

  ${IconStyle} {
    ${({ iconSize }) => iconSize && size(iconSize)}
  }
`
export default IconButtonStyle
