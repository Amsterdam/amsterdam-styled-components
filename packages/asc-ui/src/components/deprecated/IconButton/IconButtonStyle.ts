import styled from '@datapunt/asc-core'
import { size } from 'polished'
import IconStyle from '../../Icon/IconStyle'
import ButtonStyle, {
  Props as ButtonStyleProps,
} from '../../Button/ButtonStyle'

export type Props = {
  /**
   * @deprecated Please use the size prop instead.
   */
  iconSize?: number
} & ButtonStyleProps

export const IconButtonStyle = styled(ButtonStyle)<Props>`
  border: none;
  outline: none;
  padding: 0;

  ${IconStyle} {
    ${({ iconSize }) => iconSize && size(iconSize)}
  }
`
export default IconButtonStyle
