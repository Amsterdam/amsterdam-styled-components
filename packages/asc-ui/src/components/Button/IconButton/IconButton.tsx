import React from 'react'
import IconButtonStyle, {
  Props as IconButtonStyleProps,
} from './IconButtonStyle'
import Icon from '../../Icon'

type Props = {
  onClick?: React.MouseEventHandler<HTMLButtonElement>
} & IconButtonStyleProps &
  React.ButtonHTMLAttributes<HTMLButtonElement>

/**
 * @deprecated Use the Button component in combination with the Icon component
 */
const IconButton: React.FC<Props> = ({ children, onClick, ...otherProps }) => (
  <IconButtonStyle onClick={onClick} {...otherProps} square>
    <Icon>{children}</Icon>
  </IconButtonStyle>
)

export default IconButton
