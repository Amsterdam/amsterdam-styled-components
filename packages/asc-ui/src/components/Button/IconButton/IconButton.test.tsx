import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import IconButton from './IconButton'
import 'jest-styled-components'

describe('IconButton', () => {
  const onClickMock = jest.fn()
  let component: ShallowWrapper

  beforeEach(() => {
    component = shallow(
      <IconButton onClick={onClickMock}>icon-content</IconButton>,
    ).dive()
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  it('should render the IconButton', () => {
    expect(component).toMatchSnapshot()
  })

  it('should trigger onClick', () => {
    component.at(0).simulate('click')

    expect(onClickMock).toHaveBeenCalled()
  })
})
