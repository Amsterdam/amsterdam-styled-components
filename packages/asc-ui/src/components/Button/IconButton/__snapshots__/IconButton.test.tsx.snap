// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render the IconButton 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: 500;
  line-height: 20px;
  padding: ",
          [Function],
          ";
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          "transition: color 0.1s ease-in-out, background-color 0.1s ease-in-out;",
          "
  ",
          [Function],
          "
  ",
          "
  &::after {
    content: '';
    display: block;
    min-height: inherit;
    font-size: 0;
  }
",
          " // ie fix
  &:disabled {
    cursor: default;
    outline: none;
    border: none;
    color: ",
          [Function],
          ";
    background-color: ",
          [Function],
          ";
    ",
          [Function],
          ";
  }
",
          "
  border: none;
  outline: none;
  padding: 0;

  ",
          ".sc-bdVaJa",
          " {
    ",
          [Function],
          "
  }
",
        ],
      },
      "displayName": "Styled(styled.button)",
      "foldedComponentIds": Array [
        "sc-bwzfXH",
      ],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  onClick={[MockFunction]}
  square={true}
  variant="blank"
>
  <Icon>
    icon-content
  </Icon>
</StyledComponent>
`;
